src/ws/service/errors.rs

```diff
@@ -1,49 +1,35 @@
-use dcl_rpc::rpc_protocol::RemoteErrorResponse;
-use thiserror::Error;
-
-use crate::friendships::ServiceErrors;
+use crate::friendships::FriendshipServiceError;
 
+#[derive(Debug)]
 #[repr(i32)]
-#[derive(Debug, PartialEq, Eq, Error)]
-pub enum FriendshipsServiceError {
-    #[error("Unknown: {0}")]
-    Unknown(String) = ServiceErrors::Unknown as i32,
-    #[error("Bad request: {0}")]
-    BadRequest(String) = ServiceErrors::BadRequest as i32,
-    #[error("Unauthorized: {0}")]
-    Unauthorized(String) = ServiceErrors::Unauthorized as i32,
-    #[error("Forbidden: {0}")]
-    Forbidden(String) = ServiceErrors::Forbidden as i32,
-    #[error("Not found")]
-    NotFound = ServiceErrors::NotFound as i32,
-    #[error("Too many requests: {0}")]
-    TooManyRequests(String) = ServiceErrors::TooManyRequests as i32,
-    #[error("Internal server error")]
-    InternalServerError = ServiceErrors::InternalServerError as i32,
+pub enum DomainErrorCode {
+    Unknown = 0,
+    BadRequest = 400,
+    Unauthorized = 401,
+    Forbidden = 403,
+    NotFound = 404,
+    TooManyRequests = 429,
+    InternalServerError = 500,
 }
 
-impl RemoteErrorResponse for FriendshipsServiceError {
-    fn error_code(&self) -> u32 {
-        match self {
-            Self::Unknown(_) => 0,
-            Self::BadRequest(_) => 400,
-            Self::Unauthorized(_) => 401,
-            Self::Forbidden(_) => 403,
-            Self::NotFound => 404,
-            Self::TooManyRequests(_) => 429,
-            Self::InternalServerError => 500,
-        }
+pub fn as_service_error(code: DomainErrorCode, message: String) -> FriendshipServiceError {
```

Comment:
Here we can opt for using a `&str` parameter instead of an owned `String` to avoid allocations when calling this function. It's a bit more idiomatic and flexible and internally the `format!` macro will be generating a new `String` needed for the error struct creation.
```suggestion
pub fn as_service_error(code: DomainErrorCode, message: &str) -> FriendshipServiceError {
```